kolibri/core/decorators.py

```diff
@@ -275,3 +276,19 @@ def wrapped_view(*args, **kwargs):
         return view_func(*args, **kwargs)
     wrapped_view.signin_redirect_exempt = True
     return wraps(view_func)(wrapped_view)
+
+
+def cache_no_user_data(view_func):
+    """
+    Set appropriate Vary on headers on a view that specify there is
+    no user specific data being rendered in the view.
+    """
```

Comment:
how does this interact with bootstrapped data and authorization to access certain pages and data?

Reply:
This shouldn't be used on pages that bootstrap user data into the page. I have also removed most instances of bootstrapped user data in the page, because it would interfere with this.